src/test/java/com/example/cosmocats/service/ProductServiceTest.java

```diff
@@ -0,0 +1,430 @@
+package com.example.cosmocats.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.example.cosmocats.domain.Product;
+import com.example.cosmocats.dto.product.ProductDto;
+import com.example.cosmocats.dto.product.ProductListDto;
+import com.example.cosmocats.dto.product.ProductUpdateDto;
+import com.example.cosmocats.exception.ProductAlreadyExistsException;
+import com.example.cosmocats.exception.ProductNotFoundException;
+import com.example.cosmocats.repository.ProductRepository;
+import com.example.cosmocats.service.mapper.ProductMapper;
+
+import java.util.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("Product Service Tests")
+class ProductServiceTest {
+
+  @Mock private ProductRepository productRepository;
+
+  @Mock private ProductMapper productMapper;
+
+  @InjectMocks private ProductService productService;
```

Comment:
formatting again :)
basically I would prefer you to use the @SpringBootTest(class = productService.class)